src/gql_parser/campus_eatery.py

```diff
@@ -0,0 +1,295 @@
+    DescriptiveFoodItemOptionType,
+    DescriptiveFoodItemType,
+    DescriptiveFoodStationType,
+    OperatingHoursType,
+    SwipeDataType,
+)
+
+
+def get_campus_eateries(eatery_id):
+    if eatery_id is not None:
+        q = CampusEatery.query.filter_by(id=eatery_id)
+    else:
+        q = CampusEatery.query
+
+    result = conn.execute(q.statement).fetchall()
+    columns = CampusEatery.__table__.columns.keys()
+
+    populated_result = []
+    for data in result:
+        mapped_eatery = {}
+        for ind in range(len(columns)):
+            mapped_eatery[columns[ind]] = data[ind]
+        populated_eatery = parse_campus_eatery(mapped_eatery)
+        populated_result.append(populated_eatery)
+
+    return populated_result
+
+
+def parse_campus_eatery(eatery):
+    """Parses eatery data from db and populates to an object of type CampusEateryType."""
+
+    new_eatery = CampusEateryType(
+        about=eatery.get("about", ""),
+        campus_area=parse_campus_area(eatery),
+        coordinates=parse_coordinates(eatery),
+        eatery_type=eatery.get("eatery_type", ""),
+        expanded_menu=parse_expanded_menu(eatery),
+        id=eatery.get("id"),
+        image_url=eatery.get("image_url"),
+        location=eatery.get("location", ""),
+        name=eatery.get("name", ""),
+        name_short=eatery.get("name_short", ""),
+        operating_hours=parse_operating_hours(eatery),
+        payment_methods=parse_payment_methods(eatery),
+        payment_methods_enums=parse_payment_methods_enum(eatery),
+        phone=eatery.get("phone", "N/A"),
+        slug=eatery.get("slug", ""),
+        swipe_data=parse_swipe_data(eatery),
+    )
+    return new_eatery
+
+
+def parse_campus_area(eatery):
+    """Parses the common name location of an eatery.
+
+    Returns a new CampusAreaType that contains a description of an eatery
+    """
+    campus_area = eatery.get("campus_area_desc", "")
+
+    return CampusAreaType(description_short=campus_area)
+
+
+def parse_expanded_menu(eatery):
+    # query for all menu STATIONs that will be needed to populate expanded menu
+    q_stations = ExpandedMenuStation.query.filter_by(campus_eatery_id=eatery.get("id"))
+    result_stations = conn.execute(q_stations.statement).fetchall()
+    column_stations = ExpandedMenuStation.__table__.columns.keys()
+
+    menu_to_station = {}
+    station_ids = []
+    for station in result_stations:
+        mapped_station = {}
+        for ind in range(len(column_stations)):
+            mapped_station[column_stations[ind]] = station[ind]
+
+        menu_category = mapped_station["menu_category"]
+        if menu_category in menu_to_station:
+            menu_to_station[menu_category].append(mapped_station)
+        else:
+            menu_to_station[menu_category] = [mapped_station]
+        station_ids.append(mapped_station["id"])
+
+    # query for all menu ITEMs that will be needed to populate expanded menu
+    result_items = []
+    while len(station_ids) > 999:
```

Comment:
SQLite doesn't accept more than 999 variables in a single query. Doing the `filter..in` thorws errors because `station_ids` would exceed the length so I run a while loop to break up the query into multiple smaller queries and aggregate the result into a single array.